src/Utils.js

```diff
@@ -80,7 +81,12 @@ export function nodeEqual(a, b) {
       return false;
     }
   }
-  return leftKeys.length === Object.keys(right).length;
+
+  if (typeof a !== 'string' && typeof a !== 'number') {
+    return leftKeys.length === Object.keys(right).length;
```

Comment:
@ljharb I just noticed this, but should we use a lib for `Object.keys` instead of assuming it's available?